pkg/router/template/template_helper.go

```diff
@@ -0,0 +1,143 @@
+package templaterouter
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/golang/glog"
+
+	routeapi "github.com/openshift/origin/pkg/route/api"
+)
+
+func env(name, defaultValue string) string {
+	if envValue := os.Getenv(name); envValue != "" {
+		return envValue
+	}
+
+	return defaultValue
+}
+
+func isInteger(s string) bool {
+	_, err := strconv.Atoi(s)
+	return (err == nil)
+}
+
+func isTrue(s string) bool {
+	v, _ := strconv.ParseBool(s)
+	return v
+}
+
+func matchValues(s string, allowedValues ...string) bool {
```

Comment:
Could you add the logging to matchValues similar to that of the helper functions you added